### PR TITLE
Feature/fix test hanging too long

### PR DESCRIPTION
**Pull request by Dartlexx**

### CODE REVIEW

File: vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/AndroidDeviceTestRunner.kt [hunk 1]
```diff
@@ -10,46 +10,40 @@ import com.malinskiy.marathon.android.AndroidConfiguration
...
-                listener: ITestRunListener) {
+    fun execute(
+        configuration: Configuration,
+        rawTestBatch: TestBatch,
+        listener: ITestRunListener
+    ) {
+        val ignoredTests = rawTestBatch.tests.filter { it.metaProperties.contains(JUNIT_IGNORE_META_PROPERTY) }
 
-        val ignoredTests = rawTestBatch.tests.filter { it.metaProperties.contains(JUNIT_IGNORE_META_PROPERY) }
-        val testBatch = TestBatch(rawTestBatch.tests - ignoredTests)
```
Comment (memega) [inline vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/AndroidDeviceTestRunner.kt:52]: Are you sure this is an appropriate place to remove ignored tests from execution at all? Shouldn't they be handled on actor level?

File: vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/AndroidDeviceTestRunner.kt [hunk 2]
```diff
@@ -10,46 +10,40 @@ import com.malinskiy.marathon.android.AndroidConfiguration
...
-
-val JUNIT_IGNORE_META_PROPERY = MetaProperty("org.junit.Ignore")
+import kotlin.math.max
+import kotlin.math.min
 
 class AndroidDeviceTestRunner(private val device: AndroidDevice) {
 
+    companion object {
+        private val JUNIT_IGNORE_META_PROPERTY = MetaProperty("org.junit.Ignore")
+        private const val MAX_TEST_DURATION_LEEWAY = 1.2 // +20% to expected test duration
```
Comment (memega) [inline vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/AndroidDeviceTestRunner.kt:31]: These two look like good candidates for a configuration property.

File: vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/AndroidDeviceTestRunner.kt [hunk 3]
```diff
@@ -103,27 +97,38 @@ class AndroidDeviceTestRunner(private val device: AndroidDevice) {
...
+    ): RemoteAndroidTestRunner {
         val tests = testBatch.tests.map {
             "${it.pkg}.${it.clazz}#${it.method}"
         }.toTypedArray()
 
-        logger.debug { "tests = ${tests.toList()}" }
+        logger.debug { "device = [${device.serialNumber}]; tests = [${tests.toList()}]" }
 
+        // Single test timeout shouldn't be less then 'testOutputTimeoutMillis' and shouldn't be more than 'testBatchTimeoutMillis'
+        val testTimeout = min(
```
Comment (memega) [inline vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/AndroidDeviceTestRunner.kt:113]: This makes sense only if tests do not produce _any_ output at all until completion, is that really so?

**Review by memega — APPROVED** [review]

File: vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/AndroidDeviceTestRunner.kt (hunk 1 above)
Comment (Dartlexx) [reply vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/AndroidDeviceTestRunner.kt:52]: No, I also think this is not the right place for ignoring such tests. But that is out of scope for this PR and it was not changed in my PR.

File: vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/AndroidDeviceTestRunner.kt (hunk 2 above)
Comment (Dartlexx) [reply vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/AndroidDeviceTestRunner.kt:31]: Hmm, I'm not sure right now. Actually they are not playing such a big role, as test timeout is capped at batch timeout.

File: vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/AndroidDeviceTestRunner.kt (hunk 3 above)
Comment (Dartlexx) [reply vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/AndroidDeviceTestRunner.kt:113]: From what I've seen it is true at least for tests that are the problem in the current state of Marathon. But from what I've seen no log from the device is passed to Marathon until the test fails.

**Pull request merged**